float64[] kappa_ref